slycot/math.py

```diff
@@ -18,11 +18,106 @@
 #       MA 02110-1301, USA.
 
 from . import _wrapper
-from .exceptions import raise_if_slycot_error
+from .exceptions import SlycotParameterError, raise_if_slycot_error
 
 import numpy as np
 
 
+def mb02ed(typet: str,T: np.ndarray, B: np.ndarray, n: int, k: int,  nrhs: int):
+    """ X, T = mb02ed(typet, T, B, n, k, nrhs)
+
+    Solve a system of linear equations T*X = B or X*T = B with a positive
+    definite block Toeplitz matrix T.
+
+    Parameters
+    ----------
+    typet: str
+        Specifies the type of T:
+            - 'R': T contains the first block row of an s.p.d. block Toeplitz matrix,
+                and the system X*T = B is solved.
+            - 'C': T contains the first block column of an s.p.d. block Toeplitz matrix,
+                and the system T*X = B is solved.
+        Note: the notation x / y means that x corresponds to
+                typet = 'R' and y corresponds to typet = 'C'.
+    T : ndarray
+            The leading k-by-n*k or n*k-by-k part of this array must contain the first
+            block row/column of an s.p.d. block Toeplitz matrix.
+    B : ndarray
+            The leading nrhs-by-n*k or n*k-by-nrhs part of this array must contain the
+            right-hand side matrix B.
+    n : int
+            The number of blocks in T. n >= 0.
+    k : int
+       The number of rows/columns in T, equal to the blocksize. k >= 0.
+    nrhs : int
+            The number of right-hand sides. nrhs >= 0.
+
+    Returns
+    -------
+    X : ndarray
+        Leading nrhs-by-n*k / n*k-by-nrhs part of
+        this array contains the solution matrix X.
+    T: ndarray
+        If no error is thrown  and  nrhs > 0,  then the leading
+        k-by-n*k / n*k-by-k part of this array contains the last
+        row / column of the Cholesky factor of inv(T).
+
+    Raises
+    ------
+    SlycotArithmeticError
+        :info = 1:
+            the reduction algorithm failed. The Toeplitz matrix associated
+            with T is not numerically positive definite.
+    SlycotParameterError
+        :info < 0:
+            if INFO = -i, the i-th argument had an illegal value.
```

Comment:
```
>       with pytest.raises(expected_exception=SlycotParameterError, match="k must be >= 0") as cm:
E       AssertionError: Regex pattern did not match.
E        Regex: 'k must be >= 0'
E        Input: '\nif INFO = -i, the i-th argument had an illegal value.'
```

If you want to have more meaningful error messages (not mandatory);

```suggestion
    SlycotParameterError
        :info = -2:
            k must be >= 0
        :info = -3:
            n must be >= 0
```

and so on

Reply:
Otherwise just remove the `info < 0` zero case and make sure the `arg_list` is in the correct order, then `raise_if_slycot_error` will generate a reasonable message. You will have to adjust the regex pattern in the tests then.